cluster/images/etcd/migrate-if-needed.sh

```diff
@@ -46,7 +46,7 @@ set -o nounset
 BUNDLED_VERSIONS="3.0.17, 3.1.12, 3.2.24, 3.3.17, 3.4.7"
 
 # shellcheck disable=SC2039
-ETCD_NAME="${ETCD_NAME:-etcd-$HOSTNAME}"
+ETCD_NAME="${ETCD_NAME:-etcd-${ETCD_HOSTNAME:-$(hostname -s)}}"
```

Comment:
FTR - these two lines should be kept in sync:
https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L1597

@wenjiaswe - as a follow up PR, you may want to add comments both here and there that these lines should be kept in sync